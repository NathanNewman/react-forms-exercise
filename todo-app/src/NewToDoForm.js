import React, { useState } from "react";

const NewToDoForm = ({ add }) => {
  const INITIAL_STATE = {
    name: "",
  };

  const [toDo, setToDo] = useState(INITIAL_STATE);
  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setToDo({
      ...toDo,
      [name]: value,
    });
  };
  const handleSubmit = (evt) => {
    evt.preventDefault();
    add(toDo.name);
  };
  return (
    <form>
      <label htmlFor="toDoInput">Create New ToDo:</label>
      <input
        id="toDoInput"
        type="text"
        name="name"
        placeholder="ToDo"
        value={toDo.name}
        onChange={handleChange}
      />
      <button onClick={handleSubmit}>Add to list</button>
    </form>
  );
};

export default NewToDoForm;
