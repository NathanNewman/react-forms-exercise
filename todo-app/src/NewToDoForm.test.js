import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup, fireEvent } from "@testing-library/react";
import NewToDoForm from "./NewToDoForm";

afterEach(cleanup);

// smoke test
it("renders without crashing", function() {
  render(<NewToDoForm />);
});

// snapshot
test("it matches snapshot", () => {
  const { asFragment } = render(<NewToDoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("creates a new todo", () => {
  const createToDo = jest.fn();
  const { getByLabelText, getByText } = render(
    <NewToDoForm add={createToDo} />
  );
  const toDoInput = getByLabelText("Create New ToDo:");
  const button = getByText("Add to list");
  fireEvent.change(toDoInput, { target: { value: "test" } });
  fireEvent.click(button);
  expect(createToDo).toHaveBeenCalledWith("test");
});
