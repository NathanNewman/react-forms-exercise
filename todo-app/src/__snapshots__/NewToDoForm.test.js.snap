// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="toDoInput"
    >
      Create New ToDo:
    </label>
    <input
      id="toDoInput"
      name="name"
      placeholder="ToDo"
      type="text"
      value=""
    />
    <button>
      Add to list
    </button>
  </form>
</DocumentFragment>
`;
